modules/javafx.graphics/src/test/java/test/com/sun/glass/ui/monocle/FramebufferY8Test.java

```diff
@@ -0,0 +1,167 @@
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package test.com.sun.glass.ui.monocle;
+
+import com.sun.glass.ui.monocle.FramebufferY8SuperShim;
+import com.sun.glass.ui.monocle.FramebufferY8Shim;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.IntBuffer;
+import java.nio.channels.Channels;
+import javax.imageio.ImageIO;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * Provides test cases for the {@code FramebufferY8} class.
+ */
+public class FramebufferY8Test {
+
+    /**
+     * An image with all 16,777,216 colors of the 24-bit RGB palette in a 4096 ×
+     * 4096 bitmap. The image was created by Naruto_64 and published on
+     * Wikimedia Commons under the Creative Commons "CC BY-SA 4.0" license.
+     *
+     * @see
+     * <a href="https://commons.wikimedia.org/wiki/File:16777216colors_diffpatt.png">
+     * File:16777216colors diffpatt.png - Wikimedia Commons</a>
+     */
+    private static final String ALL_24BIT_RGB = "16777216colors_diffpatt.png";
+
+    private static ByteBuffer bb;
+    private static int width;
+    private static int height;
+
+    /**
+     * Loads the sample image into the composition buffer held by the
+     * {@code FramebufferY8} class. This method runs only once before all of the
+     * test cases. It loads the image into the buffer as 32-bit pixels in ARGB32
+     * format.
+     *
+     * @throws IOException if an error occurs loading the image
+     */
+    @BeforeClass
+    public static void onlyOnce() throws IOException {
+        InputStream stream = FramebufferY8Test.class.getResourceAsStream(ALL_24BIT_RGB);
+        BufferedImage image = ImageIO.read(stream);
+        width = image.getWidth();
```

Comment:
running the test gives me a Nullpointerexception here -- the resource is not on the classpath?

Reply:
Thanks, Johan. When I run the graphics tests from scratch on my build machine (after _cleanAll_ and _cleanTest_), the image file is copied to the classpath as shown below. Is that the correct location for image test files?

```ShellSession
$ find . -name 16777216colors_diffpatt.png
./modules/javafx.graphics/src/test/resources/test/com/sun/glass/ui/monocle/16777216colors_diffpatt.png

$ gradle :graphics:test
    ...
> Task :graphics:compileShimsJava
    ...
> Task :graphics:compileTestJava
    ...
BUILD SUCCESSFUL in 2m 46s
138 actionable tasks: 138 executed

$ find . -name 16777216colors_diffpatt.png
./modules/javafx.graphics/build/resources/test/test/com/sun/glass/ui/monocle/16777216colors_diffpatt.png
./modules/javafx.graphics/src/test/resources/test/com/sun/glass/ui/monocle/16777216colors_diffpatt.png
```

Reply:
This is similar to what we do for other tests.

Reply:
The image is removed, thereby removing the possibility of the `NullPointerException`. It turns out that all 24-bit RGB values can be generated by just one line of code!

```Java
IntStream.range(0, WIDTH * HEIGHT).forEachOrdered(pixels::put);
```

I made other changes to FramebufferY8Test unit tests, including some quick performance *sanity tests* of my replacement RGB565 pixel conversion.